app/src/main/java/com/itsaky/androidide/activities/editor/EditorHandlerActivity.kt

```diff
@@ -418,19 +418,14 @@ open class EditorHandlerActivity : ProjectHandlerActivity(), IEditorHandler {
 
     if (unsavedFiles.isEmpty()) {
       val file = viewModel.getCurrentFile()
-      for (i in 0 until viewModel.getOpenedFileCount()) {
-        val editor = getEditorAtIndex(i)
-
-        // Index of files changes as we keep close files
-        // So we compare the files instead of index
-        if (editor != null) {
-          if (file != editor.file) {
-            closeFile(i)
-          }
-        } else {
-          log.error("Unable to save file at index:", i)
-        }
-      }
+      val editor = getCurrentEditor()
+      
+      closeAll()
+      
+      binding.editorContainer.addView(editor)
+      binding.tabs.addTab(binding.tabs.newTab().setText(file!!.name))
+      
+      viewModel.addFile(file)
```

Comment:
This is wrong. Closing all the files at once will lead to unexpected errors. For example, on receiving the file close event, the Java language server releases all cached information about the file (like the cached parse tree, symbols, etc).

This will also mess up the document related events. For example, the `closeAll` function will broadcast the close event for the current file as well. However, the current file is added again to the editor container but no `DocumentOpenEvent` is broadcasted here. In this case, the event consumers (the Java LSP, etc) will ignore further document related events for the current file.

Reply:
I hadn't stopped to think about it!

Reply:
public void tryOpenRecentOpenedFiles() {
    if (PreferencesUtils.useOpenRecentsAutomatically()) {
      try {
        viewModel.clear();
        for (File file : indexer.getList(LAST_FILES_KEY)) {
          openFile(file, false);
        }
      } catch (Throwable e) {
        DialogUtils.newErrorDialog(
            context,
            context.getString(R.string.error),
            context.getString(R.string.error_editor_opening_recent_files) + "\n\n" + e.toString());
      }
    }
  }
this code is giving NullPointerException error is why is it opening too fast?